Add check of making `.deb` source package on pull requests

thomasspriggs opened the pull request:
Add check of making `.deb` source package on pull requests

<!---
Thank you for your contribution. Please make sure your pull request fulfils all of the below requirements. If you cannot currently tick all the boxes, but would still like to create a PR, then add the label "work in progress" and assign the PR to yourself.
--->

- [ ] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [ ] My PR is restricted to a single feature or bugfix.
- [ ] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

<!---
See, e.g., https://chris.beams.io/posts/git-commit/ for general guidelines on commit messages.

If you have created commits mixing multiple features and/or unrelated white-space changes, use a sequence involving git reset and git add -p to fix this.
--->

thomasspriggs committed:
Add cmake source package configuration

thomasspriggs committed:
Add pull request check on source packaging

So that we can check that PRs do not break source packaging. This is a
separate action, so that we get the results faster based on building the
source package without compiling first.

thomasspriggs committed:
Switch from python3 to generic python dependency

hannes-steffenhagen-diffblue commented on cmake/CPackGeneratorConfig.cmake:
```diff
@@ -0,0 +1,9 @@
+# This is run once per generator, which means that generator specific logic can
+# go here.
```
👍 sounds like the right approach on principle

hannes-steffenhagen-diffblue commented:
@thomasspriggs looks like sources for source packages need to be one level deeper, e.g. at the top level of the source package there should only be a single directory containing the source code, I think the name of that is supposed to be `<package name>-<version name>`

hannes-steffenhagen-diffblue commented on cmake/CPackGeneratorConfig.cmake:
```diff
@@ -5,5 +5,5 @@ message(STATUS "Configuring for generator ${CPACK_GENERATOR} and TOPLEVEL_TAG ${
 if("${CPACK_TOPLEVEL_TAG}" STREQUAL "Linux-Source")
   message(STATUS "Adding python dependency")
-  set(CPACK_DEBIAN_PACKAGE_DEPENDS "${CPACK_DEBIAN_PACKAGE_DEPENDS},python3")
+  set(CPACK_DEBIAN_PACKAGE_DEPENDS "${CPACK_DEBIAN_PACKAGE_DEPENDS},python")
```
Can you explain this one? Is this installing both `python3` and `python2`?

thomasspriggs replied:
The `python` dependency is for python 2.7. The `python3` dependency is for python 3.8. We actually need both to pass lintian, as we have some python scripts which start with `#!/usr/bin/env python` and with `#!/usr/bin/env python3`.

thomasspriggs committed:
Depend on both python2 and python3

thomasspriggs committed:
Add node dependency

thomasspriggs committed:
Split package building and checking steps

thomasspriggs committed:
Print more information about failed checks

thomasspriggs committed:
Add maintainer name to packages

In order to satisfy `lintian` checks.

thomasspriggs committed:
Add executable permission to scripts missing it

Because it makes it easier to use these scripts and fixes a `lintian`
warning.

thomasspriggs committed:
Add missing interpreter specification in bash script

So that `lintian` can tell this is a script.

thomasspriggs committed:
Remove executable permission from plain-text files

Because they are not executable and removing the permission fixes a
lintian warning.

thomasspriggs committed:
Switch cpack generator to tgz

thomasspriggs committed:
Set original tarball name correctly

Tarball in deb source packages must follow this naming convention
